packages/assets-controllers/src/CurrencyRateController.ts

```diff
@@ -148,81 +148,65 @@ export class CurrencyRateController extends StaticIntervalPollingController<Curr
     const releaseLock = await this.mutex.acquire();
-    const { currentCurrency, currencyRates } = this.state;
-
-    let conversionDate: number | null = null;
-    let conversionRate: number | null = null;
-    let usdConversionRate: number | null = null;
-
-    // For preloaded testnets (Goerli, Sepolia) we want to fetch exchange rate for real ETH.
-    const nativeCurrencyForExchangeRate = Object.values(
-      TESTNET_TICKER_SYMBOLS,
-    ).includes(nativeCurrency)
-      ? FALL_BACK_VS_CURRENCY // ETH
-      : nativeCurrency;
-
-    let shouldUpdateState = true;
     try {
-      if (
-        currentCurrency &&
-        nativeCurrency &&
-        // if either currency is an empty string we can skip the comparison
-        // because it will result in an error from the api and ultimately
-        // a null conversionRate either way.
-        currentCurrency !== '' &&
-        nativeCurrency !== ''
-      ) {
-        const fetchExchangeRateResponse = await this.fetchExchangeRate(
-          currentCurrency,
-          nativeCurrencyForExchangeRate,
-          this.includeUsdRate,
-        );
-        conversionRate = fetchExchangeRateResponse.conversionRate;
-        usdConversionRate = fetchExchangeRateResponse.usdConversionRate;
-        conversionDate = Date.now() / 1000;
-      }
+      const { currentCurrency } = this.state;
+
+      // For preloaded testnets (Goerli, Sepolia) we want to fetch exchange rate for real ETH.
+      // Map each native currency to the symbol we want to fetch for it.
+      const testnetSymbols = Object.values(TESTNET_TICKER_SYMBOLS);
+      const nativeCurrenciesToFetch = [...new Set(nativeCurrencies)].reduce(
+        (acc, nativeCurrency) => {
+          acc[nativeCurrency] = testnetSymbols.includes(nativeCurrency)
+            ? FALL_BACK_VS_CURRENCY
+            : nativeCurrency;
+          return acc;
+        },
+        {} as Record<string, string>,
+      );
+
+      const fetchExchangeRateResponse = await this.fetchMultiExchangeRate(
+        currentCurrency,
+        Object.values(nativeCurrenciesToFetch),
+        this.includeUsdRate,
+      );
+
+      const rates = Object.entries(nativeCurrenciesToFetch).reduce(
+        (acc, [nativeCurrency, fetchedCurrency]) => {
+          const rate = fetchExchangeRateResponse[fetchedCurrency.toLowerCase()];
+          if (rate) {
+            acc[nativeCurrency] = {
+              conversionDate: Date.now() / 1000,
+              conversionRate: rate[currentCurrency.toLowerCase()],
+              usdConversionRate: rate.usd,
+            };
+          }
+          return acc;
+        },
+        {} as CurrencyRateState['currencyRates'],
+      );
+
+      this.update((state) => {
+        state.currencyRates = {
+          ...state.currencyRates,
+          ...rates,
+        };
+      });
     } catch (error) {
-      if (
-        !(
-          error instanceof Error &&
-          error.message.includes('market does not exist for this coin pair')
```

Comment:
Unlike the single endpoint, the multi endpoint does not appear to throw if the symbol does not exist.  It just ignores that particular symbol and returns the rest.